### PR TITLE
Remove position fixed from header & rename header element from div to header

### PR DESCRIPTION
**Pull request by Misiu**
I've reordered items in the layout (header above main), renamed div to the header, and removed fixed position because it is not needed with flex.

### CODE REVIEW

**Comment by Misiu** [comment]
I didn't notice that. I'll revert position fixed

**Comment by Misiu** [comment]
@balloob done. I had to add z-index to have the header above the cards.

**Comment by balloob** [comment]
So the end result of this PR is that nothing has changed? 😄

File: raw_package/static/css/esphome-2.css [hunk 1]
```diff
@@ -41,7 +41,7 @@ body {
 
 /* Layout */
 main {
-  margin-top: 64px;
+  margin-top: 56px;
```
Comment (balloob) [inline raw_package/static/css/esphome-2.css:44]: does this mean all the content is 8px closer to the header. Do we compensate that with a padding in the content ?

**Comment by Misiu** [comment]
> So the end result of this PR is that nothing has changed? 😄

I've renamed div to header and reordered the items. So a tiny change :)
I want to clean things up before I start adding CSS variables.

File: raw_package/static/css/esphome-2.css (hunk 1 above)
Comment (Misiu) [reply raw_package/static/css/esphome-2.css:44]: The padding isn't needed. The main element has space on the top (above the cards). I'll add screenshot showing before and after.

Comment (balloob) [reply raw_package/static/css/esphome-2.css:44]: I think you forgot to add the screenshot?

Comment (Misiu) [reply raw_package/static/css/esphome-2.css:44]: I'm away from my pc, will add today when I'm home.

**Comment by Misiu** [comment]
This is how the change looks like. 64px (before) and 56px (after)
https://user-images.githubusercontent.com/1741838/197403200-00e0d263-e772-4254-a1c3-212c2e6cd8f1.mp4

**Comment by balloob** [comment]
I think that 64px looks nicer as it's equal left/top/right.

**Comment by Misiu** [comment]
> I think that 64px looks nicer as it's equal left/top/right.

the grid inside esphome-devices-list is 90% width of the screen, so I don't see the consistency.
Take a look at screenshot on wider screen:
image

**Pull request merged**